src/Text.tsx

```diff
@@ -1,13 +1,28 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const RestrictedAs = {
+  inlineText: ['span', 'p', 'b', 'i', 'em', 'small', 'strong', 'u', 'code', 'q', 's', 'label', 'div'],
+} as const
+
+const StyledText = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
   ${TYPOGRAPHY};
   ${COMMON};
   ${sx};
 `
 
-export type TextProps = ComponentProps<typeof Text>
+export type TextProps = {
+  as?: typeof RestrictedAs.inlineText[number]
+} & React.ComponentProps<'span' & typeof StyledText>
+
+const Text = ({children, as = 'span', ...rest}: TextProps) => {
```

Comment:
To avoid a breaking change, we'll need to wrap this in a `React.forwardRef()` call.

Reply:
styled-components automatically handles ref forwarding. If we're wrapping the styled component, we'll need to handle forwarding refs manually.

Reply:
I see, I'll adjust this, and make note of it for future cases!

Reply:
Adjusted! 

I saw that we utilize `PolymorphicForwardRefComponent` so I opted to utilize that for the types - but would love a double check to make sure everything looks solid TS wise.